### PR TITLE
Update `flexible_space_bar_test.dart` tests for Material 3

### PR DESCRIPTION
**Pull request by TahaTesser**
Updated unit tests for `FlexibleSpaceBar` to have M2 and M3 versions.

More info in #139076

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 1]
```diff
@@ -523,12 +541,449 @@ void main() {
...
+    // 1.5 times smaller than the full width. The height of the text is the same
+    // as the font size, with 10 dps bottom margin.
+    expect(
+      tester.getRect(find.byType(Text)),
+      rectMoreOrLessEquals(Rect.fromLTRB(0, height - titleFontSize - 10, textWidth, height), epsilon: 0.0001),
+    );
+  });
+
+  // This is a regression test for https://github.com/flutter/flutter/issues/14227
+  testWidgets('Material2 - lexibleSpaceBar sets width constraints for the title', (WidgetTester tester) async {
```
Comment (bleroux) [inline packages/flutter/test/material/flexible_space_bar_test.dart:823]:
```suggestion
  testWidgets('Material2 - FlexibleSpaceBar sets width constraints for the title', (WidgetTester tester) async {
```

**Review by bleroux — APPROVED** [review]
LGTM! with one small nit

File: packages/flutter/test/material/flexible_space_bar_test.dart [hunk 2]
```diff
@@ -78,6 +79,81 @@ void main() {
...
+    expect(effectiveStyle.color?.alpha, 128); // Which is alpha of .5
+
+    // We drag up to fully collapse the space bar.
+    await tester.drag(find.byKey(dragTarget), const Offset(0, -400.0));
+    await tester.pumpAndSettle();
+
+    expect(clipRect.size.height, minExtent);
+  });
+
+  testWidgets('Material2 - FlexibleSpaceBarSettings provides settings to a FlexibleSpaceBar', (WidgetTester tester) async {
```
Comment (QuncCccccc) [inline packages/flutter/test/material/flexible_space_bar_test.dart:135]: For the newly added test specifically for Material 3, can we also add `Material3 - ` in front of the test message?

**Review by QuncCccccc — APPROVED** [review]
LGTM:) Thanks so much for continuing to add tests!

**Pull request merged**